src/VirtoCommerce.CatalogPersonalizationModule.Web/Controllers/Api/PersonalizationModuleController.cs

```diff
@@ -75,48 +78,49 @@ public IHttpActionResult GetTagsCount(string id)
                 ResponseGroup = TaggedItemResponseGroup.WithInheritedTags.ToString()
             };
 
-            var result = _searchService.SearchTaggedItems(criteria).Results.FirstOrDefault();
+            var result = (await _searchService.SearchTaggedItemsAsync(criteria)).Results.FirstOrDefault();
             var count = result?.Tags.Union(result.InheritedTags).Distinct().Count() ?? 0;
 
-            return Ok(new { count });
+            return Ok(count);
         }
 
         /// <summary>
         /// PUT: api/personalization/taggeditem
         /// </summary>
         [HttpPut]
         [Route("taggeditem")]
-        public IHttpActionResult UpdateTaggedItem(TaggedItem taggedItem)
+        [ProducesResponseType(typeof(void), StatusCodes.Status204NoContent)]
+        public async Task<ActionResult> UpdateTaggedItem([FromBody] TaggedItem taggedItem)
         {
-            _taggedItemService.SaveTaggedItems(new[] { taggedItem });
-            _taggedItemOutlineSync.SynchronizeOutlines(new[] { taggedItem });
-            return StatusCode(HttpStatusCode.NoContent);
+            await _taggedItemService.SaveTaggedItemsAsync(new[] { taggedItem });
+            await _taggedItemOutlineSync.SynchronizeOutlinesAsync(new[] { taggedItem });
+            return Ok();
         }
 
         /// <summary>
         /// POST: api/personalization/search
         /// </summary>
         [HttpPost]
         [Route("search")]
-        [ResponseType(typeof(GenericSearchResult<TaggedItem>))]
-        public IHttpActionResult Search(TaggedItemSearchCriteria criteria)
+        [ProducesResponseType(typeof(GenericSearchResult<TaggedItem>), StatusCodes.Status200OK)]
```

Comment:
Let's add a specific class for `GenericSearchResult<TaggedItem>` to have good swagger doc (means change search service interface too).

Reply:
fixed